src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Login/LoginHandler.cs

```diff
@@ -82,10 +91,20 @@ public override Task<InitializeViewResult> InitializeObjectView(InitializeViewRe
                 ConnectPermission = prototype.WindowsGrantAccess,
                 IsEnabled = !prototype.IsDisabled,
                 IsLockedOut = prototype.IsLockedOut,
-                UserMapping = new ServerLoginDatabaseUserMapping[0],
-                SecurablePermissions = prototype.SecurablePermissions
+                UserMapping = new ServerLoginDatabaseUserMapping[0]
             };
 
+            // hide permissions for system logins or if the login doesn't support querying permissions
+            if (prototype.HidePermissions || (!dataContainer.IsNewObject && login.IsSystemObject))
+            {
+                loginInfo.HidePermissions = true;
```

Comment:
Could you just have SecurablePermissions be nullable instead and then only show if it's defined on the extension side? I thought that's what we do for other similar situations.